girder/api/v1/resource.py

```diff
@@ -49,6 +51,7 @@ def __init__(self):
         self.route('GET', (':id', 'path'), self.path)
         self.route('PUT', (':id', 'timestamp'), self.setTimestamp)
         self.route('GET', ('download',), self.download)
+        self.route('GET', ('path', 'download', '+path'), self.pathDownload)
```

Comment:
This does essentially the same type of lookup as `resource/lookup`, yet the API is totally different. Could we do anything to unify this?

Reply:
@brianhelba Do you mean the difference of the API of the endpoint? The use case of mine is looking for downloading girder item/file with relative path. Passing the path as a param won't work. And, the ability to download girder resource with a relative path will benefit not just my case. Such as OBJ, mtl, image files, .shp, .shx, .dbf, and .prj files. That's the reason David implement this API.

Reply:
Instead of variable length path, couldn't the implementation be similar to these existing end points:

![image](https://user-images.githubusercontent.com/219043/55093898-afb40e80-508b-11e9-8d13-a6c8b90767b3.png)

But in the case of `/resource/path/download` it would accept an absolute path instead of an ID.  It would basically save a call to `/resource/lookup` and then `/resource/download`